core/store/src/test/java/uk/gov/gchq/gaffer/store/ContextTest.java

```diff
@@ -141,4 +143,65 @@ public void shouldSetAndGetOriginalOpChain() {
         // Then
         assertSame(opChain, context.getOriginalOpChain());
     }
+
+    @Test
+    public void shouldShallowCloneContext() {
+        // Given
+        final User user = new User("user");
+        final String testConf = "testConf";
+        final Context context = new Context.Builder()
+                .user(user)
+                .config(testConf, "testConfVal")
+                .variable("testVar", "testVarVal")
+                .build();
+
+        // When
+        Context clonedContext = context.shallowClone();
+
+        // Then
+        assertNotSame(context, clonedContext);
+        assertEquals(context.getUser(), clonedContext.getUser());
+        assertEquals(context.getExporters().toString(),
+                clonedContext.getExporters().toString());
+        assertEquals(context.getConfig(testConf), clonedContext.getConfig(testConf));
+        assertEquals(context.getVariables(), clonedContext.getVariables());
+    }
+
+    @Test
+    public void shouldAddVariables() {
+        // Given
+        final User user = new User("user");
+        final String testConf = "testConf";
+        final Context context = new Context.Builder()
+                .user(user)
+                .config(testConf, "testConfVal")
+                .build();
+        // When
+        context.setVariable("testVar", "testVarVal");
+        context.setVariable("testVar2", "testVarVal2");
+
+        // Then
+        assertTrue(!context.getVariables().isEmpty());
```

Comment:
can we change this to assertFalse and remove the not operator.  Just can be slightly confusing to skip read if you miss the not.